stacker/providers/aws/default.py

```diff
@@ -301,15 +301,16 @@ def wait_till_change_set_complete(cfn_client, change_set_id, try_count=25,
 
 def create_change_set(cfn_client, fqn, template, parameters, tags,
                       change_set_type='UPDATE', replacements_only=False,
-                      service_role=None):
+                      service_role=None, notification_arns=[]):
```

Comment:
You shouldn't use a mutable type as an argument to a method because that is instantiated when that method is parsed, not when it's executed.  Here's an example of why not:

```
>>> def blah(a, x=[]):
...     x.append(a)
...     return x
...
>>>
>>> x = blah("apple")
>>> y = blah("banana")
>>>
>>> print x
['apple', 'banana']
>>> print y
['apple', 'banana']
```

Instead go ahead and make the default `None`, and then have something like this inside the function:

```
notification_arns = notification_arns or []
```

Reply:
And actually - I'd just leave it as None here, and then handle the converting it to a list at the `generate_cloudformation_args` method, that way you can just use None everywhere that calls that method.